### PR TITLE
Fix graph visualisation incorrectly showing edge as active

### PR DESCRIPTION
**Pull request by disrupted**

### CODE REVIEW

File: backend/streams_explorer/core/services/metric_providers.py [hunk 1]
```diff
@@ -71,9 +70,18 @@ def transform(self, data: list) -> dict:
...
+
+def is_topic(node) -> bool:
+    node_type: NodeTypesEnum = node[1]["node_type"]
+    return node_type == NodeTypesEnum.TOPIC or node_type == NodeTypesEnum.ERROR_TOPIC
+
+
 class MetricProvider:
-    def __init__(self, nodes: NodeDataView):
-        self._nodes: NodeDataView = nodes
+    def __init__(self, nodes: list):
```
Comment (philipp94831) [inline backend/streams_explorer/core/services/metric_providers.py:83]: Can we specify the type of elements of the list?

File: backend/tests/test_metricprovider.py [hunk 2]
```diff
@@ -96,24 +96,26 @@ async def mock_pull_metric(_, metric: PrometheusMetric):
...
-                replicas=1,
-                replicas_available=0,
-                consumer_read_rate=64.977769,
-            ),
             Metric(
                 node_id="atm-fraud-incoming-transactions-topic",
                 messages_in=4.8,
                 messages_out=4.8,
                 topic_size=0,
+                topic_partitions=1,
```
Comment (philipp94831) [inline backend/tests/test_metricprovider.py:104]: I thought topic partitions are not yet supported?

Comment (disrupted) [reply backend/tests/test_metricprovider.py:104]: oops this is a leftover from the other branch

File: backend/streams_explorer/core/services/metric_providers.py (hunk 1 above)
Comment (disrupted) [reply backend/streams_explorer/core/services/metric_providers.py]: added typing hints

**Review by VictorKuenstler — APPROVED** [review]

**Pull request merged**